### PR TITLE
Check that programs and dependencies are effectively provided

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -7,13 +7,17 @@ wraps = get_option('wraps')
...
 
+foreach name : get_option('depnames')
+  dep = dependency(name)
+  assert(dep.type_name() == 'internal')
+endforeach
+
+foreach name : get_option('prognames')
+  prog = find_program(name)
+  assert(prog.full_path().startswith(meson.global_build_root()))
+endforeach
```
Comment (eli-schwartz) [inline meson.build:20]: This might be a `find_program('foo.py')` instead of an `executable()`

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (xclaesse) [reply meson.build:21]: It can be in source dir indeed, fixed.

**Review by neheb — APPROVED** [review]

**Comment by neheb** [comment]
Wonder why I'm seeing this now...

```
Build started at 2022-08-24T22:57:29.206128
Main binary: /usr/bin/python3
Build Options: -Dpython.install_env=auto -Dwraps=libglvnd -Ddepvars=pcre_dep -Dlibglvnd:glx=disabled -Ddepnames=egl,opengl,glesv1,glesv2 -Dprognames=
Python system: Linux
The Meson build system
Version: 0.63.0
Source dir: /home/mangix/devstuff/wrapdb
Build dir: /home/mangix/devstuff/wrapdb/_build
Build type: native build

meson.build:1:0: ERROR: Unknown options: "depvars"

::endgroup::

```

**Comment by eli-schwartz** [comment]
Try to rm -rf the `_build` directory and run it again.

**Comment by neheb** [comment]
I did the nuclear option of wiping wrapdb and recloning :). Hope it'll work now.

**Pull request merged**